plugins/headlamp-plugin/plugin-management/plugin-management.js

```diff
@@ -260,7 +290,15 @@ async function downloadExtractPlugin(URL, headlampVersion, progressCallback, sig
     progressCallback({ type: 'info', message: 'Plugin Metadata Fetched' });
   }
   const pluginName = pluginInfo.name;
+  if (!validatePluginName(pluginName)) {
+    throw new Error('Invalid plugin name');
+  }
+
   const archiveURL = pluginInfo.data['headlamp/plugin/archive-url'];
+  if (!validateArchiveURL(archiveURL)) {
```

Comment:
archiveURL can be any url pointing to a tarball. this check is failing the install command.
<img width="1511" alt="image" src="https://github.com/headlamp-k8s/headlamp/assets/22731013/f16739e6-dc6c-4923-826f-8657d10c587f">

Reply:
IMHO I think the developer should have the flexibility to host the plugin tar where they want. The credibility cannot be just associated with the source. @joaquimrocha had a similar opinion of allowing only npm.

Reply:
I'd like us to lock it down more first, and then expand as time goes on.